Shift3/boilerplate-client-react
# Implement add agency functionality

## Changes
1. Added `CreateAgencyForm`.
2. Added `CreateAgencyView`.
3. Added `createAgency` api method.
4. Added route for create agency page.

## Purpose
Admin users should be able to create new agencies.

## Pre-Testing TODOs
Have `boilerplate-server-node` running and have a seeded `admin` user account.

## Testing
1. Pull in the changes to your local copy of this branch.
2. Run `yarn start` to start the application.
3. Login using an `admin` account and click the `Agencies` nav link  in the navbar to navigate to the agency list view.
4. Confirm that there is an `ADD AGENCY` button on the list view.
5. Confirm that clicking the `ADD AGENCY` button navigates you to the create agency view and renders a form.
6. Confirm that validation errors are displayed on the form and that the `Submit` button is disabled while there are validation errors.
7. Confirm that clicking the `CANCEL` button takes you back to the list view without making any changes.
8. Confirm that adding a valid agency name and clicking the `SUBMIT` button navigates you back to the list view and that the newly added agency appears in the list view.

### Closes #283
